modules/core/include/opencv2/core/hal/intrin_neon.hpp

```diff
@@ -537,36 +544,61 @@ OPENCV_HAL_IMPL_NEON_MUL_SAT(v_uint16x8, v_uint32x4)
 inline void v_mul_expand(const v_int8x16& a, const v_int8x16& b,
                          v_int16x8& c, v_int16x8& d)
 {
+#if CV_NEON_AARCH64
+    c.val = vmull_s8(vget_low_s8(a.val), vget_low_s8(b.val));
```

Comment:
Please, leverage the existing code.
This line starting with `c.val` is completely identical.
Let's make the scope of `#ifdef` small as possible.

Same suggestion happens to many places.